### ISSUE TITLE
hybridauth module

### ISSUE THREAD

**Issue by coolman7** [signalpoint/DrupalGap#579]
hybridauth project provides automated login and register processes from 35 social platforms (including facebook, twitter, instagram, vk, etc)

https://www.drupal.org/project/hybridauth

If we have a drupalgap module for hybridauth, we could have a better register and login interface. Currently we have only facebook module on drupalgap.

**Comment by signalpoint** [comment]
@coolman7 Yes, that'd be great. This will need contributions to be implemented.

**Comment by coolman7** [comment]
I would like to help, but I don't know how to do. Can you give me some clue or starting point to implement such drupalgap module?
Thanks

**Comment by signalpoint** [comment]
@coolman7 I'd recommend learning more about how the hybridauth module works on the Drupal side, then we'd most likely need a Drupal Services based module on the Drupal and DrupalGap side, which would help the app hit the particular Service Resources on the Drupal side.

**Comment by kinmenRP** [comment]
I'm very interested in this too. Some time ago I searched on internet to see if anybody had already implemented something in js, but the only useful link I've found was this: http://stackoverflow.com/questions/16074460/combining-hybridauth-with-javascript-sdks?rq=1 Imho hybridauth wouldbe a very nice module for drupalgap...

**Comment by gs9999** [comment]
Hello Signal point....

I think I have the skills, and want to implement a Facebook, or even Google Login for use with the hybridauth module.  

I am currently using the hybridauth module on my Drupal site without any issues, and could possibly run into issues implementing the FBOAUTH module, per your current Facebook DrupalGap module, as they are doing the same thing inside the Drupal install. (i.e. keeping track of sessions and so forth).  In any case, even if they don't conflict,  it would be an overlap in the code base.

I also wanted to implement this with (https://github.com/jeduan/cordova-plugin-facebook4), as it is against Facebook policy to use OpenFB, as it is NOT making use of the Facebook SDK.  i.e. Facebook should check for a session from the Facebook mobile app, and not from the browser.  You may be getting away with this right now, but in the future, Facebook may choose to crack down and enforce their policies (which they inevitably do).

I played with this earlier in the year (March), but had to take a break.

I have the following service resource code in place to interact with Hybridauth (Instead of FBOauth), (in this patch here https://www.drupal.org/node/2389767) and the https://github.com/jeduan/cordova-plugin-facebook4 plugin to collect a token (instead of Open FB).

I am going attempt to implement this.  I have experience with using custom services in DrupalGap, however I might need a little guidance (which I can get/copy from your Facebook DrupalGap Module), on the part where the User Object comes back from the Drupal Service with session and user object, etc. and goes live within the DrupalGap environment.

Anyways, has any body else in the community implemented this yet?

Thanks in advance,

GS
